From b6c38eb1cdb7184645507b20649d429017de9f93 Mon Sep 17 00:00:00 2001
From: Trent Mick <trentm@gmail.com>
Date: Tue, 18 Oct 2016 12:08:00 -0700
Subject: [PATCH] TOOLS-1578 new sdcnode builds for CVE-2016-5180 Reviewed by:
 Josh Wilsdon <josh@wilsdon.ca> Approved by: Josh Wilsdon <josh@wilsdon.ca>

---
 nodeconfigs.json | 127 +++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 127 insertions(+)

diff --git a/nodeconfigs.json b/nodeconfigs.json
index 9e0deb9..d51e71e 100644
--- a/nodeconfigs.json
+++ b/nodeconfigs.json
@@ -1,4 +1,23 @@
 [
+    {
+        "// image": "sdc-minimal-multiarch-lts 15.4.1",
+        "image": "18b094b0-eb01-11e5-80c1-175dac7ddf02",
+        "version": "v4.6.1",
+        "configure_flags": "--dest-cpu=ia32 --with-dtrace",
+        "gcc_version_range": "4.9.x",
+        "rpath": "$ORIGIN/../lib",
+        "include_gcc_runtime_libs": true,
+        "build_tag": "gz"
+    },
+    {
+        "// image": "sdc-minimal-multiarch-lts 15.4.1",
+        "image": "18b094b0-eb01-11e5-80c1-175dac7ddf02",
+        "comment": "For use by services running in zones. Statically link to OpenSSL and zlib *for now* until OS-1294/TOOLS-130.",
+        "version": "v4.6.1",
+        "configure_flags": "--dest-cpu=ia32 --with-dtrace",
+        "build_tag": "zone"
+    },
+
     {
         "// image": "sdc-minimal-multiarch-lts 15.4.1",
         "image": "18b094b0-eb01-11e5-80c1-175dac7ddf02",
@@ -18,6 +37,58 @@
         "build_tag": "zone"
     },
 
+    {
+        "// image": "sdc-minimal-multiarch-lts 15.4.1",
+        "image": "18b094b0-eb01-11e5-80c1-175dac7ddf02",
+        "comment": "For use by services running in zones. Dynamically link to pkgsrc's openssl and zlib?",
+        "version": "v0.12.17",
+        "configure_flags": "--dest-cpu=ia32 --with-dtrace",
+        "build_tag": "zone"
+    },
+    {
+        "// image": "sdc-multiarch 13.3.1",
+        "image": "b4bdc598-8939-11e3-bea4-8341f6861379",
+        "comment": "For use by services running in zones. Dynamically link to pkgsrc's openssl and zlib?",
+        "version": "v0.12.17",
+        "configure_flags": "--dest-cpu=ia32 --with-dtrace",
+        "build_tag": "zone"
+    },
+    {
+        "// image": "sdc-multiarch 13.3.1",
+        "image": "b4bdc598-8939-11e3-bea4-8341f6861379",
+        "comment": "For use by services running in zones. Dynamically link to pkgsrc's openssl and zlib?",
+        "version": "v0.12.17",
+        "configure_flags": "--dest-cpu=x64 --with-dtrace",
+        "build_tag": "zone64"
+    },
+    {
+        "// image": "sdc-smartos 1.6.3",
+        "image": "fd2cc906-8938-11e3-beab-4359c665ac99",
+        "comment": "For use by services running in zones. Dynamically link to pkgsrc's openssl and zlib.",
+        "version": "v0.12.17",
+        "configure_flags": "--dest-cpu=ia32 --with-dtrace --shared-zlib --shared-zlib-includes=/opt/local/include --shared-zlib-libpath=/opt/local/lib",
+        "build_env": "LDFLAGS=\"-L/opt/local/lib -R/opt/local/lib\"",
+        "build_tag": "zone"
+    },
+    {
+        "// image": "sdc-smartos 1.6.3",
+        "image": "fd2cc906-8938-11e3-beab-4359c665ac99",
+        "comment": "For use by agents in the GZ. Statically link to OpenSSL and zlib *for now* until OS-1294/TOOLS-130.",
+        "version": "v0.12.17",
+        "configure_flags": "--dest-cpu=ia32 --with-dtrace",
+        "FUTURE_configure_flags": "--with-dtrace --openssl-use-sys --openssl-libpath=/lib --openssl-includes=??? --shared-zlib --shared-zlib-libpath=/lib --shared-zlib-includes=???",
+        "build_tag": "gz"
+    },
+    {
+        "// image": "sdc-base@14.2.0",
+        "image": "de411e86-548d-11e4-a4b7-3bb60478632a",
+        "comment": "For use by services running in zones. Dynamically link to pkgsrc's openssl and zlib.",
+        "version": "v0.12.17",
+        "configure_flags": "--dest-cpu=ia32 --with-dtrace --shared-zlib --shared-zlib-includes=/opt/local/include --shared-zlib-libpath=/opt/local/lib",
+        "build_env": "LDFLAGS=\"-L/opt/local/lib -R/opt/local/lib\"",
+        "build_tag": "zone"
+    },
+
     {
         "// image": "sdc-minimal-multiarch-lts 15.4.1",
         "image": "18b094b0-eb01-11e5-80c1-175dac7ddf02",
@@ -210,6 +281,62 @@
         "build_tag": "zone"
     },
 
+    {
+        "// image": "sdc-multiarch 13.3.1",
+        "image": "b4bdc598-8939-11e3-bea4-8341f6861379",
+        "comment": "For use by services running in zones. Dynamically link to pkgsrc's openssl and zlib?",
+        "version": "v0.10.48",
+        "configure_flags": "--dest-cpu=ia32 --with-dtrace",
+        "build_tag": "zone"
+    },
+    {
+        "// image": "sdc-multiarch 13.3.1",
+        "image": "b4bdc598-8939-11e3-bea4-8341f6861379",
+        "comment": "For use by services running in zones. Dynamically link to pkgsrc's openssl and zlib?",
+        "version": "v0.10.48",
+        "configure_flags": "--dest-cpu=x64 --with-dtrace",
+        "build_tag": "zone64"
+    },
+    {
+        "// image": "sdc-smartos 1.6.3",
+        "image": "fd2cc906-8938-11e3-beab-4359c665ac99",
+        "comment": "For use by services running in zones. Dynamically link to pkgsrc's openssl and zlib.",
+        "version": "v0.10.48",
+        "configure_flags": "--dest-cpu=ia32 --with-dtrace --shared-zlib --shared-zlib-includes=/opt/local/include --shared-zlib-libpath=/opt/local/lib",
+        "build_env": "LDFLAGS=\"-L/opt/local/lib -R/opt/local/lib\"",
+        "build_tag": "zone"
+    },
+    {
+        "// image": "sdc-smartos 1.6.3",
+        "image": "fd2cc906-8938-11e3-beab-4359c665ac99",
+        "comment": "For use by agents in the GZ. Statically link to OpenSSL and zlib *for now* until OS-1294/TOOLS-130.",
+        "version": "v0.10.48",
+        "configure_flags": "--dest-cpu=ia32 --with-dtrace",
+        "FUTURE_configure_flags": "--with-dtrace --openssl-use-sys --openssl-libpath=/lib --openssl-includes=??? --shared-zlib --shared-zlib-libpath=/lib --shared-zlib-includes=???",
+        "build_tag": "gz"
+    },
+    {
+        "// image": "sdc-base@14.2.0",
+        "image": "de411e86-548d-11e4-a4b7-3bb60478632a",
+        "comment": "For use by services running in zones. Dynamically link to pkgsrc's openssl and zlib.",
+        "version": "v0.10.48",
+        "configure_flags": "--dest-cpu=ia32 --with-dtrace --shared-zlib --shared-zlib-includes=/opt/local/include --shared-zlib-libpath=/opt/local/lib",
+        "build_env": "LDFLAGS=\"-L/opt/local/lib -R/opt/local/lib\"",
+        "build_tag": "zone"
+    },
+    {
+        "// image": "sdc-base@14.2.0",
+        "image": "de411e86-548d-11e4-a4b7-3bb60478632a",
+        "comment": "For use by services running in zones. Plus the DOCKER-515 ECDH cipher support for node 0.10.",
+        "version": "v0.10.48",
+        "patch_files": [
+            "patches/ecdh-ciphers-support-v0.10.patch"
+        ],
+        "configure_flags": "--dest-cpu=ia32 --with-dtrace --shared-zlib --shared-zlib-includes=/opt/local/include --shared-zlib-libpath=/opt/local/lib",
+        "build_env": "LDFLAGS=\"-L/opt/local/lib -R/opt/local/lib\"",
+        "build_tag": "zoneecdh"
+    },
+
     {
         "// image": "sdc-multiarch 13.3.1",
         "image": "b4bdc598-8939-11e3-bea4-8341f6861379",
-- 
2.21.0

